### PR TITLE
Escape (class/property/funtion/variable) names automatically if they contain space, hyphen, or other symbols

### PR DESCRIPTION
**Pull request by alphaho**
Changes for issue #369

### CODE REVIEW

File: src/test/java/com/squareup/kotlinpoet/FunSpecTest.kt [hunk 1]
```diff
@@ -347,4 +347,14 @@ class FunSpecTest {
       |}
       |""".trimMargin())
   }
+
+  @Test fun escapePunctuationInFunctionName() {
+    val funSpec = FunSpec.builder("with-hyphen")
+            .build()
```
Comment (Egorand) [inline src/test/java/com/squareup/kotlinpoet/FunSpecTest.kt]: nit: please use 4 spaces for continuation indents

File: src/test/java/com/squareup/kotlinpoet/UtilTest.kt [hunk 2]
```diff
@@ -88,6 +88,13 @@ class UtilTest {
     assertThat("".isIdentifier).isFalse()
   }
 
+  @Test fun escapeNonJavaIdentifiers() {
+    assertThat(escapeIfNotJavaIdentifier("8startWithNumber")).isEqualTo("`8startWithNumber`")
```
Comment (Egorand) [inline src/test/java/com/squareup/kotlinpoet/UtilTest.kt]: I like the tests!

**Comment by hzsweers** [comment]
Can imports be covered in this too? Just ran into that case in some internal model generation work where we have a package name like this: `import com.uber.model.core.generated.rtapi.models.object.Meta`

File: src/test/java/com/squareup/kotlinpoet/FunSpecTest.kt (hunk 1 above)
Comment (alphaho) [reply src/test/java/com/squareup/kotlinpoet/FunSpecTest.kt]: sure. let me update it accordingly

File: src/test/java/com/squareup/kotlinpoet/UtilTest.kt (hunk 2 above)
Comment (alphaho) [reply src/test/java/com/squareup/kotlinpoet/UtilTest.kt]: thanks very much =D

**Comment by alphaho** [comment]
hi @hzsweers ,  it seems the example `import com.uber.model.core.generated.rtapi.models.object.Meta` works with the original implementation, which would call escapeIfKeyword for each part of the package name.

On escaping package names with hyphens or other symbols, it should be straightforward to make the change, I'm just not quite sure whether it's a good idea or not.

**Comment by Egorand** [comment]
@alphaho there are multiple places where the continuation indent is off, I just marked the first one - could you please fix all of them?

**Comment by hzsweers** [comment]
Just tried building against master and it looks like it didn't work with my case. Will dig in more

**Comment by hzsweers** [comment]
Nevermind, the failure is due to moshi kotlin codegen using a shaded kotlinpoet. All good :+1:

**Comment by alphaho** [comment]
@Egorand oops, let me fix them all. BTW, is the checkstyle not working properly? Thought it would help to flag all the inconsistent indents.

**Comment by alphaho** [comment]
@Egorand about `TypeAliasSpec` name, `ClassName` name, and potentially something else, I would take a look at it. I initially thought all the ones that call `escapeIfKeyword` would be the places needed to be updated.

**Comment by Egorand** [comment]
@alphaho Thanks! There's no checkstyle setup for this project, we should probably fix this, or change CONTRIBUTING.md, as it's misleading at the moment

**Comment by Egorand** [comment]
> @Egorand about TypeAliasSpec name, ClassName name, and potentially something else, I would take a look at it. I initially thought all the ones that call escapeIfKeyword would be the places needed to be updated.

Sounds good! As I said, it's fine if those are in small follow up PRs

**Comment by Egorand** [comment]
Thank you!

**Comment by alphaho** [comment]
You are welcome :-)

**Pull request merged**